plugins/jenkins_status_checker/app/models/jenkins_status_checker.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+require 'jenkins_api_client'
+require 'singleton'
+# checks jenkins staging status and publishes unstable builds checklist
+class JenkinsStatusChecker
+  include Singleton
+
+  def checklist
+    items = []
+    results = check_jenkins
+
+    if results.nil?
+      items << "Failed to retrieve jenkins checklist"
+    else
+      results["jobs"].each do |job|
+        items << "#{job["name"]} is not stable" if job["color"] != "blue"
+      end
+    end
+    items << "All projects stable!" if items.empty?
+    items
+  end
+
+  private
+
+  def check_jenkins
```

Comment:
lots of code/tests here that has very little usability impact, would be better to show the errors to the user

```
def check_jenkins
  client.get
rescue
  "Error: #{$!}"
end

...

if result.is_a?(String) # error
  items << error
else
```